project2/project2/src/menu.c

```diff
@@ -2,4 +2,94 @@
 
 void menu_start(struct state *s)
 {
+    s->editing = 0;
+    s->next_menu = menu_controller;
 }
+
+void menu_controller(struct state *s)
+{
+    switch (s->key_pressed)
```

Comment:
This will run continuously for as long as the button is held down. The tick is every 5ms, so if the user holds the button down for longer than 5ms, it will skip around.